### PR TITLE
modify: optimize

### PR DESCRIPTION
**Pull request by su-chang**
* sync room or contact before message emit;

* add dedupApi to process dupicate request for load contact / room / room member;

* add application/xml case for messageSendFile, and specify error log for sendFile timeout;

## I'm submitting a...

```
[ ] Bug Fix
[ ] Feature
[ ] Other (Refactoring, Added tests, Documentation, ...)
```

## Checklist

- [ ] Commit Messages follow the [Conventional Commits](https://conventionalcommits.org/) pattern
  - A feature commit message is prefixed "feat:"
  - A bugfix commit message is prefixed "fix:"
- [ ] Tests for the changes have been added

## Description

_please describe the changes that you are making_

_for features, please describe how to use the new feature_

_please include a reference to an existing issue, if applicable_

## Does this PR introduce a breaking change?

```
[ ] Yes
[ ] No
```

### CODE REVIEW

File: src/padplus-manager/api-request/dedupeApi.ts [hunk 1]
```diff
@@ -0,0 +1,143 @@
+import { DelayQueueExecutor } from 'rx-queue'
+import { log } from '../../config'
+import { ApiType } from '../../server-manager/proto-ts/PadPlusServer_pb'
+
+// Expire time for api call data that persist in the pool
+// Number of seconds
+const EXPIRE_TIME = 10
+
+const DEDUPE_API = [
+  ApiType.ADD_CONTACT,
```
Comment (windmemory) [inline src/padplus-manager/api-request/dedupeApi.ts]: Why do we need to dedupe this api?

Comment (su-chang) [reply src/padplus-manager/api-request/dedupeApi.ts]: It could only request once per 10 seconds, maybe useful for limit user operation.

Comment (windmemory) [reply src/padplus-manager/api-request/dedupeApi.ts]: The logic should be that, the same request (same api with same parameters) can only be emit once in every 10 seconds. If this is the case, you . can not limit user operation if the user try to add multiple person at the same time.

Comment (su-chang) [reply src/padplus-manager/api-request/dedupeApi.ts]: OK, I will remove this type.

**Pull request merged**